src/deferred.js

```diff
@@ -13,6 +13,23 @@ function Thrower( ex ) {
 	throw ex;
 }
 
+function adoptValue( value, resolve, reject ) {
+	var method;
+
+	// Check for promise aspect first to privilege synchronous behavior
+	if ( value && jQuery.isFunction( ( method = value.promise ) ) ) {
```

Comment:
![you-da-man](https://cloud.githubusercontent.com/assets/945528/14922586/1b10f02a-0e41-11e6-9a21-56b687f997e0.jpg)